test/cases/api.sh

```diff
@@ -0,0 +1,169 @@
+
+# org_id 000001 (wrong org_id)
+AuthString1="eyJlbnRpdGxlbWVudHMiOnsiaW5zaWdodHMiOnsiaXNfZW50aXRsZWQiOnRydWV9LCJzbWFydF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlfSwib3BlbnNoaWZ0Ijp7ImlzX2VudGl0bGVkIjp0cnVlfSwiaHlicmlkIjp7ImlzX2VudGl0bGVkIjp0cnVlfSwibWlncmF0aW9ucyI6eyJpc19lbnRpdGxlZCI6dHJ1ZX0sImFuc2libGUiOnsiaXNfZW50aXRsZWQiOnRydWV9fSwiaWRlbnRpdHkiOnsiYWNjb3VudF9udW1iZXIiOiIwMDAwMDAiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJ1c2VyIiwiZW1haWwiOiJ1c2VyQHVzZXIudXNlciIsImZpcnN0X25hbWUiOiJ1c2VyIiwibGFzdF9uYW1lIjoidXNlciIsImlzX2FjdGl2ZSI6dHJ1ZSwiaXNfb3JnX2FkbWluIjp0cnVlLCJpc19pbnRlcm5hbCI6dHJ1ZSwibG9jYWxlIjoiZW4tVVMifSwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMDAwMDAxIn19fQ=="
+
+CurlCmd='curl -s -w %{http_code}'
+Header="x-rh-identity: $AuthString0"
+Address="localhost"
+Port="8086"
+Version="1.0"
+MajorVersion="1"
+BaseUrl="http://$Address:$Port/api/image-builder/v$Version"
+BaseUrlMajorVersion="http://$Address:$Port/api/image-builder/v$MajorVersion"
+
+function GetResponse() {
+  arr=($1)
+  echo "${arr[@]::${#arr[@]}-1}"
+}
+
+function GetExitCode() {
+  arr=($1)
+  echo "${arr[-1]}"
+}
+
+### Case: Get version
+function Test_GetVersion() {
+  result=$($CurlCmd -H "$Header" "$url/version")
+  ver="$(GetResponse "$result" | jq .version -r)"
+  [[ $ver == "$Version" ]]
+  exit_code=$(GetExitCode "$result")
+  [[ $exit_code == 200 ]]
+}
+
+### Case: Get openapi.json
+function Test_GetOpenapi() {
+  result=$($CurlCmd -H "$Header" "$url/openapi.json")
+  exit_code=$(GetExitCode "$result")
+  [[ $exit_code == 200 ]]
+}
+
+
+### Case: Post to composer
+function Test_PostToComposer() {
+  #
+  # Prepare a request to be sent to the composer API.
+  #
+
+  REQUEST_FILE="${WORKDIR}/request.json"
+  ARCH=$(uname -m)
+  SNAPSHOT_NAME=$(uuidgen)
+
+  case $(set +x; . /etc/os-release; echo "$ID-$VERSION_ID") in
+    "rhel-8.4")
+      DISTRO="rhel-84"
+    ;;
+    "rhel-8.2" | "rhel-8.3")
+      DISTRO="rhel-8"
+    ;;
+    "fedora-32")
+      DISTRO="fedora-32"
+    ;;
+    "fedora-33")
+      DISTRO="fedora-33"
+    ;;
+    "centos-8")
+      DISTRO="centos-8"
+    ;;
+  esac
+
+  cat > "$REQUEST_FILE" << EOF
+{
+  "distribution": "$DISTRO",
+  "customizations": {
+    "packages": [
+      "postgresql"
+    ]
+  },
+  "image_requests": [
+    {
+      "architecture": "$ARCH",
+      "image_type": "ami",
+      "repositories": $(jq ".\"$ARCH\"" /usr/share/tests/osbuild-composer/repositories/"$DISTRO".json),
+      "upload_requests": [
+        {
+          "type": "aws",
+          "options": {
```

Comment:
The options for the aws upload_request only contains `share_with_accounts`, see https://github.com/osbuild/image-builder/blob/main/internal/server/api.yaml#L286

Reply:
Updated. Please help to review if the new request is correct. Thanks!